logging/kit/client_interceptors_test.go

```diff
@@ -0,0 +1,188 @@
+package kit_test
+
+import (
+	"io"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	grpc_kit "github.com/grpc-ecosystem/go-grpc-middleware/logging/kit"
+	pb_testproto "github.com/grpc-ecosystem/go-grpc-middleware/testing/testproto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+func customClientCodeToLevel(c codes.Code, logger log.Logger) log.Logger {
+	if c == codes.Unauthenticated {
+		// Make this a special case for tests, and an error.
+		return level.Error(logger)
+	}
+	return grpc_kit.DefaultClientCodeToLevel(c, logger)
+}
+
+func TestKitClientSuite(t *testing.T) {
+	if strings.HasPrefix(runtime.Version(), "go1.7") {
+		t.Skipf("Skipping due to json.RawMessage incompatibility with go1.7")
+		return
+	}
```

Comment:
Surely we don't need this check. We should only be supporting Go 1.12 and 1.13 at this point.